daemon/container_unix.go

```diff
@@ -728,7 +728,9 @@ func (container *Container) updateNetworkSettings(n libnetwork.Network) error {
 	if container.NetworkSettings == nil {
 		container.NetworkSettings = &network.Settings{Networks: make(map[string]*network.EndpointSettings)}
 	}
-
+	if !container.hostConfig.NetworkMode.IsHost() && runconfig.NetworkMode(n.Type()).IsHost() {
```

Comment:
@mrjana  yes, I able to connect to host if I disconnect the container with private network from `bridge`
<pre><code>docker run -d --name c1 busybox top
docker network disconnect bridge c1
docker network connect host c1
docker inspect c1
  ....
"Networks": {
            "host": {
                "EndpointID": "6b09beacac0470bed008ee62c55f4161701c64d5a7ccaddd8a155eb0c779bf5f",
                "Gateway": "",
                "IPAddress": "",
                "IPPrefixLen": 0,
                "IPv6Gateway": "",
                "GlobalIPv6Address": "",
                "GlobalIPv6PrefixLen": 0,
                "MacAddress": ""
            }
        }
</code></pre>
because if we disconnect it from `bridge`,  the `"NetworkSettings"` is nil and the check below for `IsPrivate()` has no chance to run